packages/agents/src/mcp/index.ts

```diff
@@ -65,13 +65,12 @@ export abstract class McpAgent<
    * Helpers
    */
 
-  async setInitialized() {
-    // TODO: move to sync api once https://github.com/cloudflare/workerd/pull/4895 lands
-    await this.ctx.storage.put("initialized", true);
+  setInitializeRequest(initializeRequest: JSONRPCMessage) {
+    this.ctx.storage.kv.put("initializeRequest", initializeRequest);
   }
 
-  async isInitialized() {
-    return (await this.ctx.storage.get("initialized")) === true;
+  getInitializeRequest() {
+    return this.ctx.storage.kv.get<JSONRPCMessage>("initializeRequest");
```

Comment:
this will require an update of runtime, might break older code 
can we use the older version for now? we can update later